File: structure/thirds.js


const thirds = arr => {
    let third = Math.floor(arr.length / 3);
    
    for (let x = 0; x < 3; x++) {
        if (x == 0) {
            let first = arr.slice(0, third).reverse();
            console.log(first);
        } else if (x == 1) {
            let second = arr.slice(third, third * 2).reverse();
            console.log(second);
        } else {
            let three = arr.slice(third * 2).reverse();
            console.log(three);
        }
    }
}

let a = [11, 45, 8, 23, 14, 12, 78, 45, 89];

thirds(a);

File: structure/thirds.py
import math 

def thirds(arr):
    third = int(math.floor(len(arr) / 3))
    x = 0
    first = []
    second = []
    three = []

    while x < 3:
        if x == 0:
            first = arr[:third]
            first = first[::-1]
            print(first)
        elif x == 1:
            second = arr[third: third * 2]
            second = second[::-1]
            print(second)
        else:
            three = arr[third * 2:]
            three = three[::-1]
            print(three)
        x += 1

    
l = [11, 45, 8, 23, 14, 12, 78, 45, 89]

thirds(l)
